operators/pkg/instance-controller/controller.go

```diff
@@ -143,17 +150,21 @@ func (r *InstanceReconciler) generateEnvironments(template *crownlabsv1alpha2.Te
 				return ctrl.Result{}, err
 			}
 		case crownlabsv1alpha2.ClassContainer:
-			return ctrl.Result{}, errors.NewBadRequest("Container Environments are not implemented")
+			if err := r.CreateContainerEnvironment(instance, &template.Spec.EnvironmentList[i], namespace, name, vmstart); err != nil {
+				return ctrl.Result{}, err
+			}
 		}
 	}
 	return ctrl.Result{}, nil
 }
 
 // SetupWithManager registers a new controller for Instance resources.
+// Also Deployments are watched in order to better handle container environment
```

Comment:
Would avoid this comment here. Probably it would be better placed just above the Owns call.